internal/gitprovider/gitprovider.go

```diff
@@ -0,0 +1,61 @@
+package gitprovider
```

Comment:
🤔 I could see the entire contents of this package being folded into the existing `git` package.

That package, already has the likes of `git.Repo`, and I think that `git.Provider` and `git.ProviderRegistry`, being the generic things they are, fit neatly in there.

Just a thought aimed at combatting the sprawl of many micro-packages.

Not a hill I would die on though.

Reply:
I think I agree with this, But I'd like to revisit combing this when we extract this out. I think providers may be a subpackage of a git package we split out.

Reply:
> I think providers may be a subpackage

I agree with that. Interfaces in the existing git package, sub-packages for specific provider implementations.